### ISSUE TITLE
Ground textures flickering 

### ISSUE THREAD

**Issue by Tristans-sky** [spirthack/CSGOSimple#58]
When i inject the dll and join a game the ground textures are flickering.
Any fix or is this a known issue?

**Comment by JimmyBassy** [comment]
It's a known issue however it's fixeable look at this thread in unknowncheats, https://www.unknowncheats.me/forum/counterstrike-global-offensive/289470-texture-glitching-createstateblock-bug-fix.html

**Comment by anarh1st47** [comment]
Fixed after https://github.com/spirthack/CSGOSimple/pull/66 ?

**Comment by rraggerr** [comment]
> Fixed after #66 ?

это не полный фикс, при некоторых обстоятельствах будет 'отвал' изза панорамы

save перед всеми действиями, restore собственно при рендере, пулл реквест делать лень так шо вы сами как нибудь


```
IDirect3DVertexDeclaration9* vert_dec; 
IDirect3DVertexShader9* vert_shader;
DWORD dwOld_D3DRS_COLORWRITEENABLE;

void save( IDirect3DDevice9* device )
{
	device->GetRenderState( D3DRS_COLORWRITEENABLE , &dwOld_D3DRS_COLORWRITEENABLE );
	device->GetVertexDeclaration( &vert_dec );
	device->GetVertexShader( &vert_shader );
	device->SetRenderState( D3DRS_COLORWRITEENABLE , 0xffffffff );
	device->SetRenderState( D3DRS_SRGBWRITEENABLE , false );
	device->SetSamplerState( NULL , D3DSAMP_ADDRESSU , D3DTADDRESS_WRAP );
	device->SetSamplerState( NULL , D3DSAMP_ADDRESSV , D3DTADDRESS_WRAP );
	device->SetSamplerState( NULL , D3DSAMP_ADDRESSW , D3DTADDRESS_WRAP );
	device->SetSamplerState( NULL , D3DSAMP_SRGBTEXTURE , NULL );
}

void restore( IDirect3DDevice9* device )
{
	device->SetRenderState( D3DRS_COLORWRITEENABLE , dwOld_D3DRS_COLORWRITEENABLE );
	device->SetRenderState( D3DRS_SRGBWRITEENABLE , true );
	device->SetVertexDeclaration( vert_dec );
	device->SetVertexShader( vert_shader );
}

**Comment by anarh1st47** [comment]
thanks, added
